search6/src/main/java/org/hibernate/performance/search/model/application/ModelServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+
+import org.hibernate.Session;
+import org.hibernate.search.engine.search.predicate.SearchPredicate;
+import org.hibernate.search.engine.search.predicate.dsl.MatchPredicateOptionsStep;
+import org.hibernate.search.engine.search.projection.SearchProjection;
+import org.hibernate.search.engine.search.projection.dsl.FieldProjectionValueStep;
+import org.hibernate.search.mapper.orm.Search;
+import org.hibernate.search.mapper.orm.automaticindexing.AutomaticIndexingStrategyName;
+import org.hibernate.search.mapper.orm.automaticindexing.session.AutomaticIndexingSynchronizationStrategyNames;
+import org.hibernate.search.mapper.orm.cfg.HibernateOrmMapperSettings;
+import org.hibernate.search.mapper.orm.common.EntityReference;
+import org.hibernate.search.mapper.orm.schema.management.SchemaManagementStrategyName;
+import org.hibernate.search.mapper.orm.scope.SearchScope;
+import org.hibernate.search.mapper.orm.session.SearchSession;
+import org.hibernate.search.util.common.data.RangeBoundInclusion;
+
+public class ModelServiceImpl implements ModelService {
+
+	@Override
+	public Properties properties(Kind kind) {
+		Properties config = new Properties();
+		config.put( HibernateOrmMapperSettings.SCHEMA_MANAGEMENT_STRATEGY,
+				SchemaManagementStrategyName.DROP_AND_CREATE_AND_DROP );
+		config.put( HibernateOrmMapperSettings.MAPPING_CONFIGURER, new SearchProgrammaticMapping() );
+		config.put( HibernateOrmMapperSettings.AUTOMATIC_INDEXING_SYNCHRONIZATION_STRATEGY,
+				AutomaticIndexingSynchronizationStrategyNames.WRITE_SYNC );
+
+		if ( kind.isLucene() ) {
+			config.put( "hibernate.search.backend.directory.type", "local-heap" );
+		}
+
+		if ( kind.isManual() ) {
+			config.put( HibernateOrmMapperSettings.AUTOMATIC_INDEXING_STRATEGY, AutomaticIndexingStrategyName.NONE );
+		}
+
+		return config;
+	}
+
+	@Override
+	public <E> List<E> search(Session session, Class<E> entityClass, Integer limit) {
+		return Search.session( session ).search( entityClass ).where( f -> f.matchAll() ).fetchHits( limit );
+	}
+
+	@Override
+	public <E> List<E> search(Session session, Class<E> entityClass, String fieldName, Object value) {
+		return Search.session( session ).search( entityClass ).where(
+				f -> f.match().field( fieldName ).matching( value ) ).fetchHits( DEFAULT_LIMIT );
+	}
+
+	@Override
+	public <E> List<E> searchAnd(Session session, Class<E> entityClass, String fieldName1, Object value1,
+			String fieldName2, Object value2) {
+		SearchSession searchSession = Search.session( session );
+
+		SearchScope<E> scope = searchSession.scope( entityClass );
+		MatchPredicateOptionsStep<?> matching1 = scope.predicate().match().field( fieldName1 ).matching( value1 );
+		MatchPredicateOptionsStep<?> matching2 = scope.predicate().match().field( fieldName2 ).matching( value2 );
+		SearchPredicate predicate = scope.predicate().bool().must( matching1 ).must( matching2 ).toPredicate();
+
+		return searchSession.search( scope ).where( predicate ).fetchHits( DEFAULT_LIMIT );
```

Comment:
This would be easier on the eye:

```suggestion
		return Search.session( session ).search( entityClass )
				.where( f -> f.bool()
						.must( f.match().field( fieldName1 ).matching( value1 ) )
						.must( f.match().field( fieldName2 ).matching( value2 ) ) )
				.fetchHits( DEFAULT_LIMIT );
```